src/app/actions/actions.tsx

```diff
@@ -0,0 +1,45 @@
+'use server'
+ 
+import { prisma } from '../../server/db';
+import { revalidatePath } from 'next/cache';
+
+interface ProjectData {
+    id?: string;
+    name: string;
+    upworkId?: string;
+    hubstaffId?: string;
+    asanaId?: string;
+}
```

Comment:
No need to write a type, infer it from your schema: https://github.com/colinhacks/zod#type-inference